pybamm/discretisations/pybamm_meshes.py

```diff
@@ -106,28 +93,50 @@ def add_ghost_meshes(self):
             edges = submesh.edges
             # left ghost cell: two edges, one node, to the left of existing submesh
             lgs_edges = np.array([2 * edges[0] - edges[1], edges[0]])
-            self[submeshname + "_left ghost cell"] = self.submeshclass(lgs_edges)
+            left_ghost_mesh = PybammSubMesh()
+            left_ghost_mesh.set_submesh(lgs_edges)
+            self[submeshname + "_left ghost cell"] = left_ghost_mesh
             # right ghost cell: two edges, one node, to the right of existing submesh
             rgs_edges = np.array([edges[-1], 2 * edges[-1] - edges[-2]])
-            self[submeshname + "_right ghost cell"] = self.submeshclass(rgs_edges)
+            right_ghost_mesh = PybammSubMesh()
+            right_ghost_mesh.set_submesh(rgs_edges)
+            self[submeshname + "_right ghost cell"] = right_ghost_mesh
 
 
-class FiniteVolumeSubmesh:
-    """A submesh for finite volumes.
+class PybammSubMesh:
+    """PyBaMM mesh class.
+        Contains the position of the nodes and the number of mesh points.
 
-    The mesh is defined by its edges; then node positions, diffs and mesh size are
-    calculated from the edge positions.
+        Parameters
+        ----------
+        domain : dict
+            A dictionary that contains the limits of the spatial variables
+        npts : dict
+            A dictionary that contains the number of points to be used on each
+            spatial variable
+        """
+
+    def set_submesh(self, edges):
+        self.edges = edges
+        self.nodes = (self.edges[1:] + self.edges[:-1]) / 2
+        self.d_edges = np.diff(self.edges)
+        self.d_nodes = np.diff(self.nodes)
+        self.npts = self.nodes.size
 
-    Parameters
-    ----------
-    edges : :class:`numpy.array`
-        The position of the edges of the cells
 
+class Pybamm1DUniformSubMesh(PybammSubMesh):
+    """
+    A class to generate a uniform submesh on a 1D domain
     """
 
-    def __init__(self, edges):
-        self.edges = edges
+    def __init__(self, lims, npts):
+
+        spatial_lims = list(lims.values())[0]
+        self.npts = list(npts.values())[0]
```

Comment:
this is ok for now but might cause problems later if `lims.values()` and `npts.values()` have more that one entry - would it be possible to pass `spatial_lims` and `self.npts` directly as arguments, instead of the whole dictionaries `lims` and `npts`? Might be easier to deal with this later when we do 2D/3D

Reply:
I agree that this will be easier to decide when we think about higher dimensions in more detail. But I think it will mostly depend on what we want the full mesh dict to look like in 2D and 3D. Do we want: 
```
mesh  = { "negative electrode": {"x": Uniform1DSubMesh(), "y": Uniform1DSubMesh()}, ...} 
```
or 
```
mesh = {"negative electrode": UniformSquare2DSubMesh(), "separator": ...}
```
UniformSquare2DSubMesh() would just contain a matrix of cell centres and cell edges instead of just an array and therefore be easy to broadcast variables onto and act on with grad and div. It also leaves open the possibility of doing other mesh shapes in 2D (e.g. triangles). This might not be the best way of doing it though. 

I wrote what is implemented with the second in mind which means that we want a general SubMesh to take in a dict of all spatial variables and their limits because that's what UniformSquare2DSubMesh would require.

Reply:
Ok happy to leave as is for now with maybe an inline comment to summarise the above